### ISSUE TITLE
Make all tests offline

### ISSUE THREAD

**Issue by jacobtomlinson** [opsdroid/opsdroid#593]
We have some tests which are checking things like pulling from GitHub, downloading files, installing deps from pypi, etc. This means the tests can break if there is no internet connection.

I've also noticed this happen from Travis where a GitHub url is denied, this may be some kind of rate limiting if lots of people on Travis are hitting GitHub.

We should mock out our tests better so they don't actually connect to the internet. This should also speed them up.

---

**Update**
Added list of failing tests so we can tick them off as they are fixed.

- [ ] tests/test_loader.py::TestLoader::test_install_default_remote_module
- [x] tests/test_loader.py::TestLoader::test_install_gist_module_file
- [x] tests/test_loader.py::TestLoader::test_install_gist_module_notebook
- [ ] tests/test_loader.py::TestLoader::test_install_specific_local_git_module
- [ ] tests/test_loader.py::TestLoader::test_install_specific_local_path_module

**Comment by FabioRosado** [comment]
I was thinking about the tests yesterday. Some of the tests I have done rely on `sleep()` which makes them slower as well. Perhaps we could make the sleep quite short (perhaps 0.1sec) so it will still get called but the delay won't be too bad, what do you think?

**Comment by athanikos** [comment]
can I take this up ?

**Comment by FabioRosado** [comment]
Yeah this issue is still open it would be great if you could work on it. Please let me know if you need any help

**Comment by Cadair** [comment]
Having just spent a good number of hours on a plane hacking on opsdroid, I can attest to this not being solved :grinning:

**Comment by Cadair** [comment]
It seems there are at least 3 config tests which still try to git pull.

**Comment by jacobtomlinson** [comment]
Sounds good.

**Comment by athanikos** [comment]
Hello, I would like to work on this if its available, thanks
